apisix/plugins/file-logger.lua

```diff
@@ -136,6 +138,20 @@ local function write_file_data(conf, log_message)
             core.log.error("failed to write file: ", conf.path, ", error info: ", err)
         end
 
+        if conf.log_to_std then
+            local std_out, err = io_open(std_out_file, "w")
```

Comment:
File open/close should be handled unified, which means you should open the stdout device in the `else` block of line #125.

Reply:
Before opening the file, we must check whether log to stdout is enabled. I am concerned that moving the file open logic above will complicate error handling.

Reply:
fixed